### DISCUSSION TITLE
Background color for Events?

### DISCUSSION THREAD

**Discussion by strommsarnac** [MMRIZE/MMM-CalendarExt3#181] [Q&A]
First off, this module is awesome, thank you for your hard work!!

So I am not a coder so I may not be using the correct terms.

I have things set up with seven (my wife is crazy...) different calendars, displaying in the same "charts".  Events from all are displaying, but not all events have a background color.  I'm not sure why, but even for the same google calendar, some events have a background color (no symbol/icon), some instead have the "symbol/icon" colored.  It's not even that all "top" events for the day have a background color, some days no events have a background color.  And some days, two events have a background color.

In config.js, in each calendars [section], each calendar has "color: "Blue,".  Different colors of course.

Any idea what I am missing?

**Comment by eouia** [comment]
You may use the `default calendar` module to provide calendar event, the default calendar module does not derive the color information of the Google calendar feed. (It is not included in the ics file format)
You can assign colors for each calendar in your config of the `default calendar`. This CX3 module will use that information on rendering.
https://docs.magicmirror.builders/modules/calendar.html#configuration-options
<img width="777" alt="image" src="https://github.com/user-attachments/assets/d883e618-33ca-4c75-a09b-3152a9495051">

**Reply by strommsarnac** [reply]
Thank you for the reply.   That's what I've done.  In the config.js file, here's what I have.
Modules: [
{
   module: "calendar",
   hiddenOnStartup: false,
   header: "Virginia's Calendar",
   position: "top_left",
   "height": "500px",
   "width": "500px",
   maximumEntries: 10, // Total Maximum Entries
   maximumNumberOfDays: 1,
   limitDays: 1, // Limit the number of days shown, 0 = no limit
   fetchInterval: 300000, // Update every five minutes
   Config: {
      calendars: [
           		{
			name: "personal",
			className: "personal",
			fetchInterval: 60 * 450,
			coloredBackground: "true",  //added as suggested, but made no difference
			color: "#0000FF",  // Blue (used Blue originally before HEX code)
			url: "https://calendar.google.com/calendar/ical/<secrethash>/basic.ics",
			},
			I have more sections for other google calendars, same format as the one above, just different color.
			]
		}	
},			
			
other modules are below the above.

See my attached image showing how it looks.

Also, I tried copying one calendar section into the calendar.js file, commenting all out from the config.js and no data displays at all.
![MM_calendar_example](https://github.com/user-attachments/assets/82591428-dc6d-4e3d-a969-8d2646cf00a0)

**Comment by strommsarnac** [comment]
Is it that the background color is only used for repeating and multi-day events?

**Reply by eouia** [reply]
Yes. A single-day event would have a transparent background and coloured symbol(or dot), and a full day or multi-day event would have a coloured background.
Of course you can override the color scheme with CSS.
